lib/datasource/remote/api/proposals_repository.dart

```diff
@@ -194,4 +194,72 @@ class ProposalsRepository extends NetworkRepository with EosRepository {
             }))
         .catchError((error) => mapEosError(error));
   }
+
+  Future<Result> setDelegate({required String accountName, required String delegateTo}) {
+    print('[eos] set delegate $accountName -> $delegateTo');
+
+    final List<Action> delegateActions = List.from(EosRepository.voiceScopes
+        .map((scope) => createDelegateAction(delegator: accountName, delegatee: delegateTo, scope: scope)));
+
+    final transaction = buildFreeTransaction(delegateActions, accountName);
+
+    return buildEosClient()
+        .pushTransaction(transaction)
+        .then((dynamic response) => mapEosResponse(response, (dynamic map) {
+              return TransactionResponse.fromJson(map);
+            }))
+        .catchError((error) => mapEosError(error));
+  }
+
+  Future<Result> undelegate({required String accountName}) {
+    print('[eos] undelegate all delegations for $accountName');
+
+    final List<Action> undelegateActions = List.from(
+        EosRepository.voiceScopes.map((scope) => createUndelegateAction(delegator: accountName, scope: scope)));
+
+    final transaction = buildFreeTransaction(undelegateActions, accountName);
+
+    return buildEosClient()
+        .pushTransaction(transaction)
+        .then((dynamic response) => mapEosResponse(response, (dynamic map) {
+              return TransactionResponse.fromJson(map);
+            }))
+        .catchError((error) => mapEosError(error));
+  }
+
+  Action createDelegateAction({
```

Comment:
make this private pls. 
This function should not be exposed as part of the repo